File: src/main/java/jk_5/nailed/gradle/delayed/DelayedFile.java
package jk_5.nailed.gradle.delayed;

import org.gradle.api.Project;

import java.io.File;

/**
 * No description given
 *
 * @author jk-5
 */
public class DelayedFile extends DelayedBase<File> {

    @SuppressWarnings({ "rawtypes", "unchecked" })
    public DelayedFile(Project owner, String pattern, IDelayedResolver... resolvers){
        super(owner, pattern, resolvers);
    }

    @Override
    public File call(){
        if (resolved == null){
            resolved = project.file(DelayedBase.resolve(pattern, project, resolvers));
        }
        return resolved;
    }

    public DelayedFileTree toZipTree(){
        return new DelayedFileTree(project, pattern, true, resolvers);
    }
}

File: src/main/scala/jk_5/nailed/gradle/delayed/DelayedFile.scala
package jk_5.nailed.gradle.delayed

import java.io.File
import org.gradle.api.Project

/**
 * No description given
 *
 * @author jk-5
 */
class DelayedFile(owner: Project, pattern: String) extends DelayedBase[File](owner, pattern) {

  override def call: File = {
    if(this.resolved.isEmpty){
      this.resolved = Option(project.file(DelayedBase.resolve(pattern, project)))
    }
    this.resolved.getOrElse(null)
  }
}
